### DISCUSSION TITLE
dotnetcoretest

### DISCUSSION THREAD

**Discussion by TerryChan2** [cake-build/cake#4112] [Q&A] [answered]
My cake version is 3.0.
I have some xunit tests in my project and would like to use dotnetcoretest in my cake script. The link (https://cakebuild.net/api/Cake.Incubator.DotNetCoreTestExtensions/DotNetCoreTestExtensions/BFD20FC8) indicated that it is a 3rd party extension provided by cake.incubator.  However, the latest cake.incubator is 7.0.0 and works only for cake 2.0 - 2.3 (https://cakebuild.net/extensions/cake-incubator/)
Any suggestions on how to resolve this?

**Comment by pascalberger** [comment, marked as answer]
If you're able to run your tests with `dotnet test` you can use the [.NET test aliases](https://cakebuild.net/dsl/dotnet/#Test) which are shipped as part of Cake.

`DotNetCoreTest` from Cake.Incubator internally calls this alias while providing a settings object specific to XUnit which is passed as arguments (see https://github.com/cake-contrib/Cake.Incubator/blob/develop/src/Cake.Incubator/DotNetCoreTestExtensions.cs#L54). If you require these parameters you can also pass them yourself to `DotNetTest` using the overload which accepts a `ProcessArgumentBuilder`: https://cakebuild.net/api/Cake.Common.Tools.DotNet/DotNetAliases/E12827F8

**Reply by TerryChan2** [reply]
Thanks for the suggestion!
